bluebubbles-server/src/server/managers/outgoingMessageManager/messagePromise.ts

```diff
@@ -64,8 +70,9 @@ export class MessagePromise {
         );
     }
 
-    resolve(value: Message | PromiseLike<Message>) {
+    async resolve(value: Message) {
         this.isResolved = true;
+        await this.updateMessageMatch(value);
         this.resolvePromise(value);
```

Comment:
I think this resolve should go before the `updateMessageMatch` call so that the API will return before the message match event gets emitted

Reply:
Done